backend/src/sdk.cpp

```diff
@@ -209,6 +240,116 @@ restinio::request_handling_status_t SDK::handleTxSDKCall(const restinio::request
     return req->create_response().set_body(absl::StrJoin(outData, ",")).done();
 }
 
+void SDK::handleSetStationState(const nlohmann::json json)
+{
+    if (!mClient->IsVoiceConnected()) {
+        TRACK_LOG_ERROR("kSetStationState requires a voice connection");
+        return;
+    }
+
+    if (!json["value"].contains("frequency")) {
+        TRACK_LOG_ERROR("kSetStationState requires a frequency");
+        return;
+    }
+
+    int frequency = json["value"]["frequency"];
+
+    mClient->SetRadioGainAll(UserSession::currentRadioGain);
+
+    if (json["value"].contains("rx")) {
+        bool oldValue = mClient->GetRxState(frequency);
+        bool newValue = Helpers::ConvertBoolOrToggleToBool(json["value"]["rx"], oldValue);
+
+        TRACK_LOG_TRACE_L1("Setting rx for {} to {}", frequency, newValue);
+        mClient->SetRx(frequency, newValue);
+
+        if (!oldValue && newValue) {
+            // When turning on RX, we refresh the transceivers
+            auto states = mClient->getRadioState();
+            if (states.find(frequency) != states.end() && !states[frequency].stationName.empty()) {
+                mClient->FetchTransceiverInfo(states[frequency].stationName);
+            }
+        }
```

Comment:
I'm not wild about how this, and other parts of this function, are a direct copy from `main.cpp`. It really seems like there should be some sort of common update function somewhere that takes an object with the new states to set, then applies them. Then both this function and `SetFrequencyState()` could use it and not duplicate code.